src/game/mod.rs

```diff
@@ -258,6 +260,79 @@ impl Game {
         Ok((num_cards, drew_monad))
     }
 
+    pub fn check_player_end(&self, player: usize) -> bool {
+        use self::card::Value::*;
+        let player = &self.players[player];
+        if ! (self.table.common.is_empty() || self.table.discard.is_empty() )  {
+            return false;
+        }
+
+        //-------------------------------------------------------------------
+
+        let trade_values: Vec<Value> =
+            (0..player.hand.len())
+                .combinations(2)
+                .filter_map(|pair| player.trade_value(pair[0], pair[1]).ok())
+                .collect();
+
+        if let Some(max_value) = trade_values.iter().max() {
```

Comment:
This logic could probably use a short explanatory comment

Reply:
I modified the logic. Let me know what you think of it--if it's clear, etc.